core/src/main/java/feign/querymap/FieldQueryMapEncoder.java

```diff
@@ -31,16 +37,20 @@
 public class FieldQueryMapEncoder implements QueryMapEncoder {
 
   private final Map<Class<?>, ObjectParamMetadata> classToMetadata =
-      new HashMap<Class<?>, ObjectParamMetadata>();
+      new ConcurrentHashMap<>();
 
   @Override
   public Map<String, Object> encode(Object object) throws EncodeException {
     try {
-      ObjectParamMetadata metadata = getMetadata(object.getClass());
-      Map<String, Object> fieldNameToValue = new HashMap<String, Object>();
+      AtomicReference<Object> atomicReference = new AtomicReference<>();
```

Comment:
I don't understand how the `atomicReference` is going to help here...

Reply:
Seconded.  This is not necessary here.  All that is required to make this thread safe is to make `ObjectParamMetadata` immutable, since that's where the information can escape.

Reply:
Hello there!
Sorry for the delay. The motivation of this pr is this https://github.com/OpenFeign/feign/issues/1257.

And the use of the atomic reference was because of this comment https://github.com/OpenFeign/feign/issues/1257#issuecomment-697046816 maybe i mis understood.

Cheers,

Reply:
`ObjectParamMetadata` is immutable. It uses a `final` "unmodifiable" list. The only issue I can see with the old code is that there could be multiple calls to generate the same metadata object at the same time (although only 1 will end up in the map) which a later PR fixes nicely.

FYI - When I wrote the old `QueryMapEncoder`, I felt that lower sync overhead later was worth a one-time possible multiple initialization slowdown.